Sources/XcbeautifyLib/Renderers/TeamCityRenderer.swift

```diff
@@ -0,0 +1,251 @@
+import Colorizer
+import Foundation
+
+struct TeamCityRenderer: OutputRendering {
+    let colored: Bool
+    let additionalLines: () -> String?
+    let terminalRenderer: TerminalRenderer
+
+    init(colored: Bool, additionalLines: @escaping () -> String?) {
+        self.colored = colored
+        self.additionalLines = additionalLines
+        self.terminalRenderer = TerminalRenderer(colored: colored, additionalLines: additionalLines)
+    }
+
+    private func outputTeamCityProblem(text: String, filePath: String) -> String {
+        """
+        ##teamcity[buildProblem description='\(filePath)']
+        \(text)
+        """
+    }
+
+    func outputTeamCityError(text: String, details: String) -> String {
+        """
+        ##teamcity[message text='\(text)' errorDetails='\(details.teamCityEscaped())' status='ERROR']
+        \(text)
+        """
+    }
+
+    func outputTeamCityWarning(text: String, details: String) -> String {
+        """
+        ##teamcity[message text='\([text, details.teamCityEscaped()].joined(separator: "|n"))' status='WARNING']
+        \(text)
+        """
+    }
+
+    func formatFailingTest(group: FailingTestCaptureGroup) -> String {
+        return self.terminalRenderer.formatFailingTest(group: group)
+    }
+
+    func formatUIFailingTest(group: UIFailingTestCaptureGroup) -> String {
+        return self.terminalRenderer.formatUIFailingTest(group: group)
+    }
+
+    func formatRestartingTest(group: RestartingTestCaptureGroup) -> String {
+        return self.terminalRenderer.formatRestartingTest(group: group)
+    }
+
+    func formatTestCasePending(group: TestCasePendingCaptureGroup) -> String {
+        return self.terminalRenderer.formatTestCasePending(group: group)
+    }
+
+    func formatTestCaseMeasured(group: TestCaseMeasuredCaptureGroup) -> String {
+        return self.terminalRenderer.formatTestCaseMeasured(group: group)
+    }
+
+    func formatTestCasePassed(group: TestCasePassedCaptureGroup) -> String {
+        return self.terminalRenderer.formatTestCasePassed(group: group)
+    }
+
+    func formatParallelTestCasePassed(group: ParallelTestCasePassedCaptureGroup) -> String {
+        return self.terminalRenderer.formatParallelTestCasePassed(group: group)
+    }
+
+    func formatParallelTestCaseAppKitPassed(group: ParallelTestCaseAppKitPassedCaptureGroup) -> String {
+        return self.terminalRenderer.formatParallelTestCaseAppKitPassed(group: group)
+    }
+
+    func formatParallelTestCaseFailed(group: ParallelTestCaseFailedCaptureGroup) -> String {
+        return self.terminalRenderer.formatParallelTestCaseFailed(group: group)
+    }
```

Comment:
Hi @dimones. These functions should use the default implementations from `OutputRendering` instead of referencing them via `TerminalRenderer`. We should delete these methods and move any missing default implementations from `TerminalRenderer` to the respective `OutputRendering` extension.